tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/beam_pipeline_test.py

```diff
@@ -25,6 +25,7 @@
 from tfx.experimental.distributed_inference.graphdef_experiments.subgraph_partitioning import create_complex_graph
 from tfx.experimental.distributed_inference.graphdef_experiments.subgraph_partitioning import graph_partition
 
+tf.compat.v1.enable_eager_execution()  # Re-enable eager mode
```

Comment:
If we need eager execution mode for _some_ tests, my strong preference and suggestion is to use a [pytest yield fixture](https://docs.pytest.org/en/stable/how-to/fixtures.html#yield-fixtures-recommended) as a decorator to enable and disable it for the tests that need it. The fixture can be stored in a `conftest.py` and reused anywhere it is needed.

CC: @peytondmurray

Reply:
:point_up: To add context to this point:

- `tfx` has a hard dependency on `tensorflow>=2.15, <2.16`. Eager execution is [enabled by default](https://www.tensorflow.org/api_docs/python/tf/compat/v1/enable_eager_execution) for `tensorflow>=2`, so I don't think this is necessary.
- This setting unilaterally applies global changes to the testing environment, impacting much more than just the tests here. Even if `tfx` did allow users to install `tensorflow<2`, I'd still vote to move this into a test fixture to limit the impact on all the other tests in the test run.

On a related note, how many folks are still running TF<2? TF 2.0 came out over 5 years ago - if `tfx` still has tests that target deprecated 1.0 functionality, it may be worth starting an issue/discussion to remove them, as I don't think `tfx` can't be expected to run these at this point.

Reply:
The issue stemmed from the create_complex_graph module, which forcibly disables eager execution upon initialization. I've added a pytest yield fixture to re-enable it when the tests using this module end.

https://github.com/tensorflow/tfx/blob/ee6eaf2c81916170571cc9288112c8a43504c3ef/tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/create_complex_graph.py#L25

Reply:
`[pytest yield fixture](https://docs.pytest.org/en/stable/how-to/fixtures.html#yield-fixtures-recommended)` is not working, since disable_eager_execution() isn't working as expected. This might be due to disable_eager_execution() being called during the collection phase of pytest, as it's also called during initialization. This code is from an older part of the project, and I'm unsure if it's safe to remove. However, since it's within experimental code, I'll comment it out for now.